scio-elasticsearch/es2/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -227,18 +284,25 @@ public void processElement(ProcessContext c) throws Exception {
       private final SerializableFunction<T, Iterable<ActionRequest<?>>> toActionRequests;
       private final ThrowingConsumer<BulkExecutionException> error;
       private final int maxBulkRequestSize;
+      private final FluentBackoff backoffConfig;
 
       public ElasticsearchWriter(String clusterName,
-                                 InetSocketAddress[] servers,
-                                 int maxBulkRequestSize,
-                                 SerializableFunction<T, Iterable<ActionRequest<?>>> toActionRequests,
-                                 ThrowingConsumer<BulkExecutionException> error) {
+          InetSocketAddress[] servers,
+          int maxBulkRequestSize,
+          SerializableFunction<T, Iterable<ActionRequest<?>>> toActionRequests,
+          ThrowingConsumer<BulkExecutionException> error, int maxRetries, int retryPause) {
         this.maxBulkRequestSize = maxBulkRequestSize;
         this.clientSupplier = new ClientSupplier(clusterName, servers);
         this.toActionRequests = toActionRequests;
         this.error = error;
+
+        this.backoffConfig = FluentBackoff.DEFAULT
+            .withMaxRetries(maxRetries)
```

Comment:
If the param were `maxAttempts`, this would make sense - but when we say "3 retries," we are effectively saying 4 attempts. I'd resolve to keep this as is.